packages/loader/container-loader/src/connectionManager.ts

```diff
@@ -403,16 +405,18 @@ export class ConnectionManager implements IConnectionManager {
         }
     }
 
-    public connect(connectionMode?: ConnectionMode) {
-        this.connectCore(connectionMode).catch((error) => {
+    public connect(connectionMode?: ConnectionMode, progress?: IProgress) {
+        this.connectCore(connectionMode, progress).catch((error) => {
             const normalizedError = normalizeError(error, { props: fatalConnectErrorProp });
             this.props.closeHandler(normalizedError);
         });
     }
 
-    private async connectCore(connectionMode?: ConnectionMode): Promise<void> {
+    private async connectCore(connectionMode?: ConnectionMode, progress?: IProgress): Promise<void> {
         assert(!this.closed, 0x26a /* "not closed" */);
 
+        this._cancelConnectSignal = progress?.cancel;
```

Comment:
You can't simply overwrite global state (this._cancelConnectSignal) here. Consider connect() called twice.
If you overwrite it, then you will screw either first or second call (depending how this global is used), but there is no way to make it work correctly that way.

You would need to do one of the two approaches:
1. Not pass it in, but return from the call, allowing connect() flow to create it on the fly or return existing one. Ok for cancelation itself, but not Ok for the rest of IProgress (retry function) as there would be no way for caller to learn about retries (shape of IProgress would need to be changed to reflect difference in data flow).
2. Implement chaining of IProgress objects. I believe I typed example earlier in this PR somewhere (at least for cancelation). Let me know if you want me to get to more details